### ISSUE TITLE
Enable adding new pages to the site

### ISSUE THREAD

**Issue by huima** [event-catalog/eventcatalog#494]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/boyney123/eventcatalog/blob/master/CONTRIBUTING.md#reporting-new-issues).

### Description

Eventhough you can add links to other sites by adding links to header or footer, it could make sense to enable the possibility to add new pages to the nextjs site by copying for example pages directory from project directory to the .eventcatalog directory in eventcatalog/bin/eventcatalog.js script.

Adding:

 fs.copySync(path.join(projectDIR, 'pages'), path.join(eventCatalogLibDir, 'pages'));

To each of the commands could be enough?

### Motivation

This would enable users to define new pages in the project directory - or even overwrite some functionality of Event Catalog pages if -- for some strange reason -- they would need to do something like that.

**Comment by boyney123** [comment]
Interesting idea, what kind of pages are you looking to add?

**Comment by alkeshdagade** [comment]
@boyney123 Not sure what @huima is looking for, however from my perspective this feature will help to customize Catalog as per business need e.g. we can have subdomains and have linkage between domain, subdomain and events.

**Comment by seanhagen** [comment]
I'm also interested in being able to do this -- being able to put things like guides, tutorials, etc into the generated site would be a huge benefit to me and my team. It would mean we'd be able to build a single place for all our event & API related documentation to live, greatly simplifying our lives 😄

> +1 I had read the customize landing page part of the docs and assumed that adding entries to /pages would do exactly this.

This is what I assumed as well, but when looking at the list of generated routes I can see the other Markdown files I put into `/pages` aren't rendered into HTML.

**Comment by carlosallexandre** [comment]
Hey everyone,

For customization, I think a better approach would be to create a Custom Loader (new Astro content layer) that leverages @eventcatalog/sdk instead of Astro's getCollection. This way, we could transform @eventcatalog/core into an Astro template, similar to [astro starlight](https://github.com/withastro/starlight), giving users full flexibility to customize their catalog as needed.

I can explore the Custom Loader further next week. Let me know if this approach sounds reasonable!

**Comment by boyney123** [comment]
Hey @carlosallexandre 


It's a good question, I need to dive deeper into this, I seen starlight and like the idea, but I also like control of the whole thing.... I think as we move forward in the future there are more things I would like to experiment with like hybrid builds, (vs static opt in), and things like this. Side car apps for schema registries etc etc, so I feel its a one way door, meaning I don't wanna go that route and be locked into the framework / template boundaries (if that makes sense).

So IMO let's hold off for now, although I do think there is some tidy up we can do with custom loaders anyway, our loders at the moment copy files over etc etc, but I wonder if custom loaders would tidy this code up or not.... but saying that,  I wonder how much value there is moving them or not.... maybe something we can look at, at some point.

**Comment by boyney123** [comment]
> I'm also interested in being able to do this -- being able to put things like guides, tutorials, etc into the generated site would be a huge benefit to me and my team. It would mean we'd be able to build a single place for all our event & API related documentation to live, greatly simplifying our lives 😄
> 
> > +1 I had read the customize landing page part of the docs and assumed that adding entries to /pages would do exactly this.
> 
> This is what I assumed as well, but when looking at the list of generated routes I can see the other Markdown files I put into `/pages` aren't rendered into HTML.

Yeah I think this makes sense tbh.... gotta figure out some kind of API etc, but I Think the simular to docusarus config would be good, where you can group your files in your eventcatalog.config file and have these render in the UI etc. This would let anyone document anything, we could introduce a new section in the left nav too to help with this, and have a place users can just add anything they want into this section.

**Comment by boyney123** [comment]
Custom docs are live https://www.eventcatalog.dev/docs/development/customization/custom-documentation/introduction

Custom homepages etc are coming.
